apps/qubit/modules/function/actions/editAction.class.php

```diff
@@ -61,7 +61,7 @@ protected function earlyExecute()
             $this->form->setWidget('serialNumber', new sfWidgetFormInputHidden());
         } else {
             // Check authorization
-            if (!QubitAcl::check($this->parent, 'create')) {
+            if (!QubitAcl::check($this->resource, 'create')) {
```

Comment:
I think this is fine.  If `QubitFunctionObject` allowed hierarchies, then it would make sense to check permissions against the parent because the resource we are creating doesn't exist yet (which is why we check against the parent for actors, information objects, etc.).  `QubitAcl` has hard-coded roles for editing/creating a function resource anyway, so any `QubitFunctionObject` (including an empty one) will produce the same results here.